Cockpit 284 (#599)
cockpit-project/cockpit-project.github.io: _posts/2023-01-25-cockpit-284.md

---
title: Cockpit 284
author: pitti
date: '2023-01-25'
tags: cockpit, machines
slug: cockpit-284
category: release

> summary: ''

Author: ```suggestion
summary: 'User logs, NBDE, forceful snapshot reverting, and improved default container command'
```

---

Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).
We release regularly.

> Here are the release notes from Cockpit 284 and cockpit-machines 282:

Author: ```suggestion
Here are the release notes from Cockpit 284, cockpit-machines 282, and cockpit-podman 284:
```

## Services: Show logs for user units

The Services page now shows journal logs for user systemd services.

![screenshot of show logs for user units]({{ site.baseurl }}/images/284-services-show-logs-for-user-units.png)

*Many thanks to [Timothy Johnson](https://github.com/t1m0thyj) for this contribution!*

> ## Storage: Set up a system to use NBDE

Author: ```suggestion
## Storage: Set up NBDE
```

---
title: Cockpit 284
author: pitti
date: '2023-01-25'
tags: cockpit, machines
slug: cockpit-284
category: release
summary: 'User logs, NBDE, forceful snapshot reverting, and improved default container command'
---

Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).
We release regularly.

Here are the release notes from Cockpit 284, cockpit-machines 282, and cockpit-podman 61:

## Services: Show logs for user units

The Services page now shows journal logs for user systemd services.

![screenshot of show logs for user units]({{ site.baseurl }}/images/284-services-show-logs-for-user-units.png)

*Many thanks to [Timothy Johnson](https://github.com/t1m0thyj) for this contribution!*

## Storage: Set up NBDE

> It is possible (and unfortunately likely) that a system is misconfigured and can not mount filesystems during boot that rely on Clevis to unlock them. Cockpit can now fix this when adding a keyserver to a encrypted filesystem.

Author: ```suggestion
A common misconfiguration of NBDE can prevent mounting filesystems that depend on Clevis. Cockpit now detects and fixes this issue while adding a keyserver to an encrypted filesystem.
```

![screenshot of set up a system to use nbde]({{ site.baseurl }}/images/284-storage-set-up-a-system-to-use-nbde.png)

## Machines: Option to forcefully revert a snapshot

> There are situations where a snapshot revert involves extra risk and the user should either consider not reverting snapshot, or force it. The few most common cases are:
> - Snapshot that lacks full domain information for reverting configuration. In this case, the user should be sure that snapshot is compatible with the current configuration (if it is not, the domain will likely fail to run).
> - Reverting from a running domain to an active state where a new hypervisor has to be created rather than reusing the existing hypervisor.
> - Snapshot of inactive VM which also has a managed saved state. This requires forceful revert because the snapshot does not contain a memory state and will therefore not replace the existing memory state. This ends up switching a disk underneath a running system and can cause filesystem corruption or crashes due to swap content mismatches when run.

Author: I think this is far too verbose. If this is really relevant and isn't addressed by the UI, then we should really try to do a follow-up. I don't think this should be documented in the release notes like this.

```suggestion
Cockpit can now revert virtual machine snapshots.
```

Reviewer: It was able to do that before, the new feature is that you can do it also in situations which the normal feature disallowed until now. I.e. UI for adding a `--force` flag.

Reviewer: ```suggestion
In some situations, reverting a snapshot involves extra risk and the user should either consider not reverting, or force it. The Machines page can now do this, and explains the reason.
```

Author: OK, this merges the two suggestions, with some revisions.

```suggestion
Cockpit can now forcefully revert virtual machine snapshots.

Reverting a snapshot may involve risk. The revert dialog will explain these corner cases with a message, to help determine if forcing a revert should be preferred or to consider not reverting a snapshot.
```
